lib/jekyll/layout.rb

```diff
@@ -43,7 +45,11 @@ def initialize(site, base, name)
     #
     # Returns nothing.
     def process(name)
-      self.ext = File.extname(name)
+      self.mime_type = MIME::Types.find{|t| ".#{t.preferred_extension}" == File.extname(name)}
```

Comment:
Does this need to use `mime_type`?
Will this work just having a bunch of extensions?

(the question started out being rhetorical, but to be honest, I don't know)